lib/sqlalchemy/sql/schema.py

```diff
@@ -5664,3 +5687,93 @@ def _copy(self, **kw: Any) -> Identity:
         )
 
         return self._schema_item_copy(i)
+
+
+class Period(SchemaItem):
+    """Defines a period column, i.e. "PERIOD FOR" syntax
+
+    See the linked documentation below for complete details.
+
+    .. versionadded:: TODO
+
+    .. seealso::
+
+        :ref:`temporal_ddl`
+    """
+
+    __visit_name__ = "period"
+
+    table: Table
+
+    def __init__(
+        self,
+        name: str,
+        start: _DDLColumnArgument,
+        end: _DDLColumnArgument,
+    ) -> None:
+        """Construct a PERIOD FOR DDL construct to accompany a
+        :class:`_schema.Column`.
+
+        :param name:
+          The name of the period
+
+        :param start:
+            A target column that defines the period's start. A
+            :class:`_schema.Column` object or a column name as a string.
+
+        :param end:
+            A target column that defines the period's start. A
+            :class:`_schema.Column` object or a column name as a string.
+        """
+        self.name = name
```

Comment:
I would leave "name" out as a public attribute, at least for Period.  users wouldn't instantiate Period directly.  Does ApplicationTimePeriod require a user-custom "name" ?

Reply:
ApplicationTimePeriod does, that can be named anything. Really, we could _probably_ drop the `ApplicationTimePeriod` class since it doesn't really mean anything separate from `Period` - application time is just a `PERIOD FOR` definition added that isn't named system_time, no special DDL syntax.

Reply:
at the moment it looks like we are going with Period and SystemTimePeriod, that's fine